README.rst

```diff
@@ -348,6 +352,73 @@ This setting takes the URL of the logo to be displayed on email notification.
 **Note**: Provide a URL which points to the logo on your own web server. Ensure that the URL provided is
 publicly accessible from the internet. Otherwise, the logo may not be displayed in email.
 
+REST API
+--------
+
+Live documentation
+~~~~~~~~~~~~~~~~~~
+
+.. image:: docs/images/api-docs.png
+
+A general live API documentation (following the OpenAPI specification) is available at ``/api/v1/docs/``.
+
+Browsable web interface
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. image:: docs/images/api-ui.png
+
+Additionally, opening any of the endpoints `listed below <#list-of-endpoints>`_
+directly in the browser will show the `browsable API interface of Django-REST-Framework
+<https://www.django-rest-framework.org/topics/browsable-api/>`_,
+which makes it even easier to find out the details of each endpoint.
+
+Authentication
+~~~~~~~~~~~~~~
+
+The API authentication is based on session based authentication via  REST framework.
```

Comment:
Why isn't this as https://github.com/openwisp/openwisp-firmware-upgrader/#authentication
How to authenticate is explained in openwisp-users to avoid duplicating the information in every module.
When we'll have the comprehensive developer documentation we'll be linking a different section of the same documentation instead of a different repository, that will be the best, but for now, linking a different repo is the best option we have (duplicating the info in all the modules is unmaintainable for us right now).

Reply:
I have used session based authentication as in [openwisp-ipam](https://github.com/openwisp/openwisp-ipam#api-authentication). I didn't thought it would be necessary to use use token based authentication.

Reply:
openwisp-ipam is not up to date, see https://github.com/openwisp/openwisp-ipam/issues/33

Base your work by following [openwisp-firmware-upgrader](https://github.com/openwisp/openwisp-firmware-upgrader), which is up to date.

And yes, please add support for bearer authentication, we want to make sure the API can be used also outside of django easily. Beware of this bug in openwisp-users: https://github.com/openwisp/openwisp-users/issues/126 @atb00ker should fix that soon, coordinate with him if you have any issue.

Reply:
@pandafy https://github.com/openwisp/openwisp-firmware-upgrader/pull/85